skimage/restoration/_denoise.py

```diff
@@ -470,3 +470,94 @@ def denoise_wavelet(img, sigma=None, wavelet='db1', mode='soft'):
 
     clip_range = (-1, 1) if img.min() < 0 else (0, 1)
     return np.clip(out, *clip_range)
+
+
+def _sigma_est_dwt(detail_coeffs, distribution='Gaussian'):
+    """
+    Calculation of the robust median estimator of the noise standard
+    deviation.
+
+    Parameters
+    ----------
+    detail_coeffs : ndarray
+        The detail coefficients corresponding to the discrete wavelet
+        transform of an image.
+    distribution : str
+        The underlying noise distribution.
+
+    Returns
+    -------
+    sigma : float
+        The estimated noise standard deviation (see section 4.2 of [1]_).
+
+    References
+    ----------
+    .. [1] D. L. Donoho and I. M. Johnstone. "Ideal spatial adaptation
+       by wavelet shrinkage." Biometrika 81.3 (1994): 425-455.
+    """
+    # consider regions with detail coefficients exactly zero to be masked out
+    detail_coeffs = detail_coeffs[np.nonzero(detail_coeffs)]
+
+    if distribution.lower() == 'gaussian':
+        # 75th quantile of the underlying, symmetric noise distribution:
+        # denom = scipy.stats.norm.ppf(0.75)
+        denom = 0.67448975019608171
+        sigma = np.median(np.abs(detail_coeffs)) / denom
+    else:
+        raise ValueError("Only Gaussian noise estimation is currently "
+                         "supported")
+    return sigma
+
+
+def estimate_sigma(im, multichannel, average_sigmas=True):
+    """
+    Robust wavelet-based estimator of the noise standard deviation.
+
+    Parameters
+    ----------
+    im : ndarray
+        Image for which to estimate the noise standard deviation.
+    multichannel : bool
```

Comment:
I agree that it is a bit inconvenient to have to type out multichannel=False every time when working with graycale images.
That said, I expect that a fairly common error would then be passing in a color image, but not setting the multichannel argument.  Maybe raise a warning if the size is 3 on the last axis and multichannel is False?